imports/ui/components/hoc/AuthenticatedRoute.jsx

```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+import { Route, Redirect } from 'react-router-dom'
+import { AuthConsumer } from './AuthProvider'
+
+const AuthenticatedRoute = ({ component: Component, ...rest }) => (
+  <AuthConsumer>
+    {({ isLoggedIn, user }) => (
+      <Route
+        render={props =>
+          isLoggedIn ? <Component user={user} {...props} /> : <Redirect to="/login" />
```

Comment:
@d3vild06 How about removing the prop `user={user}` from Component? Now user can be accessed from context API below every level.

Reply:
I am not 100% sure this will work since the `Component` being rendered from the `<Route>` component is a functional component. According to the docs, the only way you can subscribe to a context is via the [Context.Consumer](https://reactjs.org/docs/context.html#contextconsumer) API.

I will play around with this to see if this is the only way to go and update the component accordingly.

Reply:
I'm a bit confused about what's happening here. What does `Now user can be accessed from context API below every level` mean -- or can someone point me to a resource? Thanks!

Reply:
@gauravchl according to the React documentation, the only way a functional component can subscribe to a context is via the `Context.Consumer` component. However, it requires a function as a child (render prop). The passed `value` argument to that function is equal to the value prop of the closest Provider for this context above in the tree. 

In our case, that `value` param is our user from our state in our HOC. No way around this. We _must_ pass in this argument to be able to reference the user.  You can find the doc info [here](https://reactjs.org/docs/context.html#contextconsumer)

I've made some additional minor modifications to be able to fully use the context API across the existing components that require the user. I think this is ready for final review!